src/Grid.cpp

```diff
@@ -0,0 +1,322 @@
+//
+//  Grid.cpp
+//  Game of Life
+//
+//  Created by Alex Frasca on 01/05/2023.
+//
+
+#include "Grid.hpp"
+
+namespace conway {
+
+// Cell
+
+Cell::Cell(sf::Vector2<int> idx, sf::Vector2<float> pos, float spc, CELL_STATE st) {
+    index = idx;
+    position = pos;
+    spacing = spc;
+    state = st;
+    kill_next_cycle = false;
+    activate_next_cycle = false;
+    
+    //init cell drawable
+    drawable = sf::ConvexShape();
+    drawable.setPointCount(4);
+    drawable.setPoint(0, sf::Vector2f( position.x          , position.y           ));
+    drawable.setPoint(1, sf::Vector2f( position.x + spacing, position.y           ));
+    drawable.setPoint(2, sf::Vector2f( position.x + spacing, position.y + spacing ));
+    drawable.setPoint(3, sf::Vector2f( position.x          , position.y + spacing ));
+}
+
+void Cell::tick() {
+    // change colors based on state for style
+    switch (state) {
+        case CELL_STATE::CELL_OFF:
+            drawable.setOutlineThickness(0);
+            drawable.setFillColor(GoL_Black);
+            break;
+        case CELL_STATE::CELL_LOW:
+            drawable.setOutlineColor(GoL_NavyBlue);
+            drawable.setOutlineThickness(-2);
+            drawable.setFillColor(GoL_DarkFucshia);
+            break;
+        case CELL_STATE::CELL_MEDIUM:
+            drawable.setOutlineColor(GoL_DarkFucshia);
+            drawable.setOutlineThickness(-2);
```

Comment:
Is `-2` valid for outline thickness? Check SFML docs / or the type of the parameter, if it's an unsigned integer type, this argument will underflow.